execution_engine/src/engine_state/mod.rs

```diff
@@ -582,11 +524,10 @@ where
 
         // Get payment purse Key from handle payment contract
         // payment_code_spec_6: system contract validity
-        let payment_purse_key =
-            match handle_payment_named_keys.get(handle_payment::PAYMENT_PURSE_KEY) {
-                Some(key) => *key,
-                None => return Ok(ExecutionResult::precondition_failure(Error::Deploy)),
-            };
+        let Some(payment_purse_key) =
+            handle_payment_named_keys.get(handle_payment::PAYMENT_PURSE_KEY).copied() else {
+            return Ok(ExecutionResult::precondition_failure(Error::Deploy));
```

Comment:
I'll add a new variant `FailedToGetPaymentPurse` as that's what the error case is, and I don't want to remove the `Deploy` variant in case it breaks backwards compatibility (I think these errors are stored in the form of a string in storage, so probably no issue with renaming here, but with the JSON-RPC server schema test having been removed I have no confidence that that's true :) ).

Reply:
Done in [d8cbbdf](https://github.com/casper-network/casper-node/pull/4544/commits/d8cbbdfe05181fbe83a5f667364f18bd2a8ab052).